Frameworks/UIKit/UIViewController.mm

```diff
@@ -1207,43 +1196,40 @@ - (void)presentViewController:(UIViewController*)controller animated:(BOOL)anima
 
     [[UIApplication sharedApplication] beginIgnoringInteractionEvents];
 
-    if ([controller _hidesParent]) {
-        [self _notifyViewWillDisappear:animated];
-    }
-
-    priv->_presentedViewController = controller;
     [controller view];
```

Comment:
reason for setting the presentedViewController after [controller view] (which implicitly can call viewDidLoad) is to match ref platform behavior; but this is not something validated by the #2194 testcase.